client/systems.go

```diff
@@ -70,3 +74,32 @@ func (client *Client) ListSystems() ([]System, error) {
 	}
 	return rsp.Systems, nil
 }
+
+// DoSystemAction issues a request to perform an action using the given seed
+// system and its mode.
+func (client *Client) DoSystemAction(systemLabel string, action *SystemAction) error {
+	if systemLabel == "" {
+		return fmt.Errorf("cannot request an action without the system")
+	}
+	if action == nil {
+		return fmt.Errorf("cannot request an action without one")
+	}
+	// deeper verification is done by the backend
+
+	req := struct {
+		Action string `json:"action"`
+		*SystemAction
+	}{
+		Action:       "do",
+		SystemAction: action,
+	}
+
+	var body bytes.Buffer
+	if err := json.NewEncoder(&body).Encode(&req); err != nil {
+		return err
+	}
+	if _, err := client.doSync("POST", "/v2/systems/"+systemLabel, nil, nil, &body, nil); err != nil {
```

Comment:
Some wondering... Should we do some basic validation of systemLabel here? Out of curiosity I've checked existing client code and this lead to interesting errors ;), e.g.
```$ snap change ////
error: cannot unmarshal: json: cannot unmarshal array into Go value of type client.changeAndData
```

Reply:
Hmmm, valid concern. I'll do a followup with checks in the `seed` package.

Reply:
This is done in #8358